packages/react/src/ErrorBoundary.tsx

```diff
@@ -36,8 +36,7 @@ export interface ErrorBoundaryFallbackProps<TError extends Error = Error> {
 type EnabledCallback = (error: Error) => boolean
 type Enabled = ConstructorType<Error> | EnabledCallback
 const shouldCatch = (error: Error, enabled: Enabled) =>
-  (enabled.prototype instanceof Error && error instanceof enabled) ||
-  (typeof enabled === 'function' && (enabled as EnabledCallback)(error))
+  enabled.prototype instanceof Error ? error instanceof enabled : (enabled as EnabledCallback)(error)
```

Comment:
I checked if enabled(ErrorConstructor) extend Error correctly. after it enabled will be EnabledCallback.